new_dawn_server/users/api/tests/test_update_profile.py

```diff
@@ -75,25 +120,26 @@ def test_register_update_user(self):
         all_arguments_2 = {
             **register_arguments_2,
             **account_arguments_2,
-            **profile_arguments_2
-        }
-        all_arguments_2 = {
-            "objects": [all_arguments_2]
+            **profile_arguments_2,
         }
 
         self.api_client.put(
-            "/api/v1/register/", format="json", data=all_arguments_2)
+            "/api/v1/register/1/", format="json", data=all_arguments_2)
 
-        self.assertEqual(User.objects.count(), 1)
-        self.assertEqual(Account.objects.count(), 1)
-        self.assertEqual(Profile.objects.count(), 1)
-        self.assertEqual(User.objects.first().first_name, "test-update")
-        self.assertEqual(Account.objects.first().gender, "F")
-        self.assertEqual(Profile.objects.first().description, "nice22222")
-        self.assertEqual(Profile.objects.first().degree, "high school")
+        self.assertEqual(User.objects.count(), 2)
+        self.assertEqual(Account.objects.count(), 2)
+        self.assertEqual(Profile.objects.count(), 2)
+        self.assertEqual(User.objects.get(id=1).first_name, "test-change")
+        self.assertEqual(Account.objects.get(user_id=1).gender, "F")
+        self.assertEqual(Profile.objects.get(user_id=1).description, "nice22222")
+        self.assertEqual(Profile.objects.get(user_id=1).degree, "high school")
 
         res = self.api_client.get("/api/v1/profile/", format="json")
         res_data = json.loads(res.content)
-        user_data = res_data["objects"][0]
+        user_data = res_data["objects"][1]
```

Comment:
Why user 1 is mapped to object 1? Should it be object 0?

Reply:
Because original first profile was deleted, when you call get profile, if returns according to the order of creation of profile.